[docs] EAS Update documentation organization pass (#31359)
expo/expo: docs/pages/eas-update/getting-started.mdx

</Step>

<Step label="7">
## Publish an update

Publishing an update allows:

- Fixing bugs and quickly updating non-native parts of a project instead of creating a new build
- [Sharing a preview version of an app](/review/overview/) using internal distribution

To publish an update with changes from your project, you can use the `eas update` command, and specify a name for the `branch` and a `message` to describe the update:

<Terminal cmd={['$ eas update --branch [branch-name] --message "[message]"']} />

<Collapsible summary="How does publishing an update work?">

> When you publish an update with the `eas update` command, it generates a new update bundle and uploads it to the EAS servers. The `branch` name is used to identify the update and is associated with the uploaded bundle. It is similar to how Git commit works, where every commit is associated with a Git branch.

Reviewer: ```suggestion
When you publish an update with the `eas update` command, it generates a new update bundle and uploads it to the EAS servers. The `branch` name is used to identify which bucket of updates to put the new one in. It is similar to how Git commit works, where every commit is associated with a Git branch.
```

(my grammar is bad. but for our current data model I tend to describe branches as ordered streams of updates, and a channel is a pointer into that stream)

> For example, you want to publish an update for the [`development` build profile](/build/eas-json/#build-profiles). You can use it as the branch name to associate the update with its build profile. When the app requests an update, it will download the update associated with the `development` branch.

Reviewer: s/branch/channel ?

> For example, you want to publish an update for the [`development` build profile](/build/eas-json/#build-profiles). You can use it as the branch name to associate the update with its build profile. When the app requests an update, it will download the update associated with the `development` branch.

Author: ```suggestion
For example, you want to publish an update for the [`development` build profile](/build/eas-json/#build-profiles). You can use it as the channel name to associate the update with its build profile. When the app requests an update, it will download the update associated with the `development` channel.
```

> For this example, the command to publish an update with the `development` branch and a message is:

Author: ```suggestion
For this example, the command to publish an update with the `development` channel and a message is:
```

> <Terminal cmd={['$ eas update --branch development --message "Fixes typo"']} />

Author: ```suggestion
<Terminal cmd={['$ eas update --channel development --message "Fixes typo"']} />
```

<BoxLink
  title="In-depth guide on how EAS Update works"
  description="Dive deep and learn how EAS Update works."
  href="/eas-update/how-it-works/"
  Icon={LayersTwo02Icon}
/>

</Collapsible>

</Step>

<Step label="8">
## Test the update

After the update is uploaded to EAS Update, you can use one of the following methods to test the update:

- Use the Extensions tab in a [development build](/eas-update/expo-dev-client/) to load the update.
- Use [Expo Orbit](/review/with-orbit/) to install and launch the update in a development build.
- Implement a custom strategy with [Updates API](/versions/latest/sdk/updates/) and [app config](/versions/latest/config/app/#updates) to load updates in the app programmatically.
- Manually test the update by force closing and reopening a release build of your app up to two times to download and apply the update. Updates for non-development builds (preview or production) are automatically downloaded to the device in the background when the app starts up and makes a request for any new updates. The updated will be applied after it is downloaded and the app is restarted.

<Collapsible summary="Something not working?">

> If your app is not updating as expected, see the following debugging guides for techniques to validate your configuration:

Reviewer: add link

> If your app is not updating as expected, see the following debugging guides for techniques to validate your configuration:

Author: ```suggestion
If your app is not updating as expected, see the [debugging guide](/eas-update/debug/) for techniques to validate your configuration.
```
